packages/atomic/src/components/commerce/atomic-commerce-recommendation-list/atomic-commerce-recommendation-list.spec.ts

```diff
@@ -0,0 +1,834 @@
+import {buildFakeRecommendations} from '@/vitest-utils/testing-helpers/fixtures/headless/commerce/recommendations-controller.js';
+import {buildFakeSummary} from '@/vitest-utils/testing-helpers/fixtures/headless/commerce/summary-subcontroller.js';
+import {
+  buildRecommendations,
+  InteractiveProduct,
+  InteractiveProductProps,
+  Product,
+} from '@coveo/headless/commerce';
+import {page} from '@vitest/browser/context';
+import '@vitest/browser/matchers.d.ts';
+import {html} from 'lit';
+import {describe, expect, vi, beforeEach, it} from 'vitest';
+import {
+  ItemDisplayBasicLayout,
+  ItemDisplayDensity,
+  ItemDisplayImageSize,
+} from '../../common/layout/display-options.js';
+// TODO: Replace with atomic-commerce-recommendation-interface bindings once it is merged (KIT-3934)
+import {CommerceBindings} from '../atomic-commerce-interface/atomic-commerce-interface.js';
+import './atomic-commerce-recommendation-list';
+import {AtomicCommerceRecommendationList} from './atomic-commerce-recommendation-list.js';
+
+vi.mock('@/src/components/common/interface/store', {spy: true});
+vi.mock('@coveo/headless/commerce', {spy: true});
+
+describe('AtomicCommerceRecommendationList', () => {
+  const interactiveProduct = vi.fn();
+  const promoteChildToParent = vi.fn();
+  const summary = vi.fn();
+
+  beforeEach(() => {
+    fixtureCleanup();
+
+    summary.mockReturnValue(buildFakeSummary());
+
+    vi.mocked(buildRecommendations).mockReturnValue(
+      buildFakeRecommendations({
+        implementation: {
+          interactiveProduct,
+          promoteChildToParent,
+          summary,
+        },
+      })
+    );
+  });
+
+  const setupElement = async ({
+    display = 'grid',
+    density = 'normal',
+    imageSize = 'small',
+    productsPerPage = 3,
+    isAppLoaded = true,
+  }: {
+    display?: ItemDisplayBasicLayout;
+    density?: ItemDisplayDensity;
+    imageSize?: ItemDisplayImageSize;
+    productsPerPage?: number;
+    isAppLoaded?: boolean;
+  } = {}) => {
+    const {element} =
+      await renderInAtomicCommerceRecommendationInterface<AtomicCommerceRecommendationList>(
+        {
+          template: html`<atomic-commerce-recommendation-list
+            .display=${display}
+            .density=${density}
+            .imageSize=${imageSize}
+            .productsPerPage=${productsPerPage}
+          ></atomic-commerce-recommendation-list>`,
+          selector: 'atomic-commerce-recommendation-list',
+          bindings: (
+            bindings: Partial<CommerceBindings> & typeof defaultBindings
+          ) => {
+            bindings.store.state.loadingFlags = isAppLoaded
+              ? []
+              : ['loading-flag'];
+            bindings.engine.logger = {error: vi.fn()} as never;
+            return bindings;
+          },
+        }
+      );
+
+    return element;
+  };
+
+  const getParts = (element: AtomicCommerceRecommendationList) => {
```

Comment:
Aren't we missing a few parts here?

- label
- previous-button
- next-button
- indicators
- indicator
- active-indicator

I see you're testing those parts, though, so it would just be a matter of defining them here instead of inline in the tests. At this point it doesn't matter that much, your choice :)

also, the `outline` part isn't documented in the component.